### PR TITLE
Adding a new Arabic dataset 

### PR DESCRIPTION
**Pull request by Ruqyai**
## Checklist for adding MMTEB dataset

Reason for dataset addition:

- [ ] I have tested that the dataset runs with the `mteb` package.
- [ ] I have run the following models on the task (adding the results to the pr). These can be run using the `mteb -m {model_name} -t {task_name}` command.
  - [ ] `sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2`
  - [ ] `intfloat/multilingual-e5-small`
- [ ] I have checked that the performance is neither trivial (both models gain close to perfect scores) nor random (both models gain close to random scores).
- [ ] If the dataset is too big (e.g. >2048 examples), considering using `self.stratified_subsampling() under dataset_transform()`
- [ ] I have filled out the metadata object in the dataset file (find documentation on it [here](https://github.com/embeddings-benchmark/mteb/blob/main/docs/adding_a_dataset.md#2-creating-the-metadata-object)).
- [ ] Run tests locally to make sure nothing is broken using `make test`. 
- [ ] Run the formatter to format the code using `make lint`. 
- [ ] I have added points for my submission to the [points folder](https://github.com/embeddings-benchmark/mteb/blob/main/docs/mmteb/points.md) using the PR number as the filename (e.g. `438.jsonl`).

### CODE REVIEW

**Comment by Ruqyai** [comment]
hello dears
I faced this error, could anyone help me
`..../datasets/builder.py", line 599, in _create_builder_config
    raise ValueError(
ValueError: BuilderConfig 'corpus' not found. Available: ['default']`

**Comment by Ruqyai** [comment]
I saw @thakur-nandan's PR on https://github.com/embeddings-benchmark/mteb/pull/642, then I took snippets of his code to see if they would solve my problem.

**Review by imenelydiaker — COMMENTED** [review]
Hey @Ruqyai,
Thank you for this addition! Can you please run both models [intfloat/multilingual-e5-small](https://huggingface.co/intfloat/multilingual-e5-small) and [sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2](https://huggingface.co/sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2)?
Also run `make pr` or `make lint` locally and push changes?

File: mteb/tasks/InstructionRetrieval/ara/WikipediaInstructionRetrieval.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+
+
+class WikipediaInstructionRetrieval(AbsTaskInstructionRetrieval):
+    metadata = TaskMetadata(
+        name="WikipediaInstructionRetrieval",
+        description="This dataset contains a pre-processed version from Wikipedia suitable for semantic search.",
+        reference="https://huggingface.co/datasets/Cohere/wikipedia-22-12-ar-embeddings",
+        dataset={
+            "path": "Cohere/wikipedia-22-12-ar-embeddings",
```
Comment (imenelydiaker) [inline mteb/tasks/InstructionRetrieval/ara/WikipediaInstructionRetrieval.py:89]: I'm not sure how this dataset can be used for retrieval. It doesn't contain queries according to what's on HF, I only see pargraph ids, doc ids and text content but no queries.
I think we're missing the queries here 🤔

Comment (imenelydiaker) [reply mteb/tasks/InstructionRetrieval/ara/WikipediaInstructionRetrieval.py:89]: There are these two datasets although it's part of miracl, which we already handle: 
image

**Comment by Ruqyai** [comment]
> Hey @Ruqyai, Thank you for this addition! Can you please run both models [intfloat/multilingual-e5-small](https://huggingface.co/intfloat/multilingual-e5-small) and [sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2](https://huggingface.co/sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2)? Also run `make pr` or `make lint` locally and push changes?

I am trying but I face this error:
`dist-packages/datasets/builder.py", line 599, in _create_builder_config
    raise ValueError(
ValueError: BuilderConfig 'corpus-ar' not found. Available: ['default']`

**Comment by imenelydiaker** [comment]
> > Hey @Ruqyai, Thank you for this addition! Can you please run both models [intfloat/multilingual-e5-small](https://huggingface.co/intfloat/multilingual-e5-small) and [sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2](https://huggingface.co/sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2)? Also run `make pr` or `make lint` locally and push changes?
> 
> I am trying but I face this error: `dist-packages/datasets/builder.py", line 599, in _create_builder_config raise ValueError( ValueError: BuilderConfig 'corpus-ar' not found. Available: ['default']`

It is caused by this code: 
```py
  corpus_identifier = f"corpus-{lang}"
  corpus_data = datasets.load_dataset(
       path,
       corpus_identifier,
       cache_dir=cache_dir,
       revision=revision,
       trust_remote_code=True,
  )
```

The dataset doesn't have a subset corpus-ar, the only available subset is the one by default.

Just as I mentionned in my previous comment this dataset is not suitable for retrieval.

**Pull request closed without merging**